src/routers/posts.ts

```diff
@@ -22,25 +22,1077 @@ import {
+ *                     items:
+ *                       type: string
+ *                     description: Array of user IDs who liked the post.
+ *                   creator_id:
+ *                     type: string
+ *                     description: The user ID of the post creator.
+ *       500:
+ *         description: Internal server error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 err:
+ *                   type: string
+ *                   description: Error message
+ */
 router.get('/', getAllPosts);
+
+/**
+ * @swagger
+ * /posts/getPostByID/{postID}:
+ *   get:
+ *     tags:
+ *       - Posts operations
+ *     summary: Get a post by its ID
+ *     description: Retrieve a specific post by providing its ID.
+ *     parameters:
+ *       - in: path
+ *         name: postID
+ *         required: true
+ *         schema:
+ *           type: string
+ *         description: The unique identifier of the post.
+ *     responses:
+ *       200:
+ *         description: A post object.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 _id:
+ *                   type: string
+ *                   description: The unique identifier of the post.
+ *                 title:
+ *                   type: string
+ *                   description: The title of the post.
+ *                 content:
+ *                   type: string
+ *                   description: The content of the post.
+ *                 createdAt:
+ *                   type: string
+ *                   format: date-time
+ *                   description: The creation timestamp of the post.
+ *                 creator_id:
+ *                   type: object
+ *                   properties:
+ *                     _id:
+ *                       type: string
+ *                       description: The unique identifier of the creator.
+ *                     username:
+ *                       type: string
+ *                       description: The username of the post creator.
+ *       404:
+ *         description: Post not found
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 err:
+ *                   type: string
+ *                   description: Error message
+ *       500:
+ *         description: Internal server error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 err:
+ *                   type: string
+ *                   description: Error message
+ */
 router.get('/getPostByID/:postID', getPostByID);
+
+/**
+ * @swagger
+ * /posts/count:
+ *   get:
+ *     tags:
+ *       - Posts operations
+ *     summary: Count all posts
```

Comment:
summary should contains just -
"Must be connected as admin" OR
"Must be connected as user"